doc/man3/OSSL_CMP_ITAV_set0.pod

```diff
@@ -45,21 +96,109 @@ the I<itav> as generic B<ASN1_TYPE> pointer.
 OSSL_CMP_ITAV_push0_stack_item() pushes I<itav> to the stack pointed to
 by I<*itav_sk_p>. It creates a new stack if I<*itav_sk_p> points to NULL.
 
+OSSL_CMP_ITAV_new0_caCerts() creates a new B<OSSL_CMP_ITAV> structure
+of type B<caCerts> that includes the optionally given list of certificates.
+
+OSSL_CMP_ITAV_get0_caCerts() assigns to I<*out> an internal pointer to
+the list of CA certificates contained in the infoValue field of I<itav>.
+The pointer may be NULL if no CA certificate is included or
```

Comment:
The pointer also might not be touched. IMO better semantics is to not touch outputs on error, or rather keep the output value undefined in case of error to keep a leeway for future changes.

Reply:
I had thought it would be beneficial to initialize the values as far as possible even on error as a precaution, but now I've switched to the design principle you mention here,
(of course, both in the code, and in the documentation).